src/main/update/check.ts

```diff
@@ -0,0 +1,112 @@
+import { IApv, ISimpleApv } from "src/interfaces/apv";
+import { IDownloadUrls, getDownloadUrls } from "../../utils/url";
+import Headless from "../headless/headless";
+
+export class GetPeersApvFailedError extends Error {}
+
+export interface IUpdateContext {
+  newApv: ISimpleApv;
+  oldApv: ISimpleApv;
+  urls: IDownloadUrls;
+}
+
+export async function checkUpdateRequired(
+  standalone: Headless,
+  platform: NodeJS.Platform,
+  // TODO: should make config object
+  netenv: string,
+  peerInfos: string[],
+  baseUrl: string,
+  localApvToken: string,
+  trustedApvSigners: string[]
+): Promise<IUpdateContext | null> {
+  let peersApv;
+  try {
+    peersApv = await getPeersApv(standalone, peerInfos, trustedApvSigners);
+  } catch (e) {
+    console.error(`getPeersApv Error ocurred ${e}:\n`, e.stderr);
+    return null;
+  }
+
+  const localApv = await getLocalApv(standalone, localApvToken);
+
+  if (updateRequired(peersApv.version, localApv.version))
+    return {
+      newApv: peersApv,
+      oldApv: localApv,
+      urls: getDownloadUrls(baseUrl, netenv, peersApv.version, platform),
+    };
+
+  return null;
+}
+
+// FIXME: Could use overload function...
+export async function checkUpdateRequiredUsedPeersApv(
+  peersApv: ISimpleApv,
+  standalone: Headless,
+  platform: NodeJS.Platform,
+  netenv: string,
+  baseUrl: string,
+  localApvToken: string
+): Promise<IUpdateContext | null> {
+  const localApv = await getLocalApv(standalone, localApvToken);
+
+  if (updateRequired(peersApv.version, localApv.version))
+    return {
+      newApv: peersApv,
+      oldApv: localApv,
+      urls: getDownloadUrls(baseUrl, netenv, peersApv.version, platform),
+    };
+
+  return null;
+}
+
+export function checkCompatible(
+  peersApv: ISimpleApv,
+  localApv: ISimpleApv
+): boolean {
+  const peersCompatVersion = BigInt(
+    (peersApv.extra["CompatiblityVersion"] as string | number) ?? 0
+  );
+  const localCompatVersion = BigInt(
+    (localApv.extra["CompatiblityVersion"] as string | number) ?? 0
+  );
+
+  if (peersCompatVersion > localCompatVersion) return false;
+
+  return true;
+}
+
+async function getPeersApv(
```

Comment:
This and `getLocalApv` doesn't do anything async, although I think the standalone functions should've been async functions.

Reply:
Remove async